docs/guides/scaffolding.md

```diff
@@ -0,0 +1,121 @@
+Developing with the {{open_autonomy}} framework usually entails creating all the packages that the new agents need: skills, connections, protocols... This can seem a bit overwhelming at the beginning, as new developers might not be familiar with the structure of each one of those. To simplify the development, the framework provides a scaffold tool that lets developers speed up their flow by auto-generating boilerplate code.
+
+##What you will learn
+
+In this guide, we will show how to:
+
+- Create different packages using the scaffold tool.
+
+## Setup
+
+!!!note
+    You might have already prepared your environment if you went through the [quick start](https://docs.autonolas.network/quick_start/) guide. If that's the case, you can skip this section.
+
+Before starting this guide, ensure that your machine satisfies the [framework requirements](./quick_start.md#requirements) and that
+you have followed the [setup instructions](./quick_start.md#setup). As a result you should have a Pipenv workspace folder.
+
+1. Create a workspace folder:
+```bash
+mkdir my_workspace
+cd my_workspace
+```
+
+2. Setup the environment. Remember to use the Python version you have installed. Here we are using 3.10 as reference:
+```bash
+touch Pipfile && pipenv --python 3.10 && pipenv shell
+```
+
+3. Install the {{open_autonomy}} framework:
+```bash
+pip install open-autonomy
+```
+
+4. Initialize the framework to work with the remote [IPFS](https://ipfs.io) registry. This means that when the framework will be fetching a component, it will do so from the [IPFS](https://ipfs.io):
+    ```bash
+    autonomy init --remote --ipfs
+    ```
+
+## Using the scaffold tool: step-by-step instructions
+
+First, create an empty agent using the autonomy CLI. This will create the agent directory, that will contain folders for connections, contracts, protocols and skills.
+```
+autonomy create my_agent
+cd my_agent
+```
+Now that the agent is in place, you can follow the next sections examples to add your packages.
+
+### Scaffold a connection.
+To add your first [connection](https://open-aea.docs.autonolas.tech/connection/), run:
+```
+autonomy scaffold connection my_connection
+```
+
+### Scaffold a contract.
+To scaffold a new [contract](https://open-aea.docs.autonolas.tech/contract/), run:
+```
+autonomy scaffold contract my_contract
+```
+
+### Scaffold a protocol.
+To generate a new [protocol](https://open-aea.docs.autonolas.tech/protocol/), execute:
+```
+autonomy scaffold protocol my_protocol
+```
+
+### Scaffold a skill.
+Create a simple [skill](https://open-aea.docs.autonolas.tech/skill/) running:
+```
+autonomy scaffold skill my_skill
+```
```

Comment:
Remove all this - irrelevant for 99%

Reply:
wasn't this supposed to be about scaffolding? If I remove all scaffold commands, what's the point of the PR? What should be left then?

Reply:
Only FSM scaffolding?

Reply:
Yes which is what I asked for since two weeks and which is the only relevant scaffolding command in the context of open-autonomy for now.